### PR TITLE
Skip YAML binding for SSL integration test that doesn't need it

### PR DESCRIPTION
**Pull request by dsyer**
Shaves about 1.5sec off the test suite (out of 13).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -47,12 +53,6 @@
 			<artifactId>ehcache</artifactId>
 		</dependency>
 
-		<dependency>
```
Comment (snicoll) [inline pom.xml]: Did you intend to remove devtools?

File: src/test/java/io/spring/start/site/StartApplicationHttpsTests.java [hunk 2]
```diff
@@ -41,7 +41,9 @@
  * @author Stephane Nicoll
  */
 @RunWith(SpringJUnit4ClassRunner.class)
-@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT, properties = "initializr.env.force-ssl=true")
+@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT, properties = {
+		"initializr.env.force-ssl=true",
+		"spring.config.location=file:./src/test/resources/application-test.yml" })
```
Comment (snicoll) [inline src/test/java/io/spring/start/site/StartApplicationHttpsTests.java]: This means that tests have to be ran from the module directory. This sounds kind of obvious but I don't like the idea of hardcode relative paths here.

File: src/test/resources/application-test.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+local:
```
Comment (snicoll) [inline src/test/resources/application-test.yml]: I don't understand the content of this file. The `gcp.version` seems unrelated to the task at hand. Same thing for logging & co.

File: pom.xml (hunk 1 above)
Comment (dsyer) [reply pom.xml]: No, I guess not. Sorry.

File: src/test/java/io/spring/start/site/StartApplicationHttpsTests.java (hunk 2 above)
Comment (dsyer) [reply src/test/java/io/spring/start/site/StartApplicationHttpsTests.java]: Where else would they be run from? We can use `classpath:...` I suppose. Want me to change it?

File: src/test/resources/application-test.yml (hunk 3 above)
Comment (dsyer) [reply src/test/resources/application-test.yml]: It's just the `application.yml` without the `initializr.*` properties. There didn't seem a lot of point fretting about the rest, since they are pretty fast to bind.

Comment (snicoll) [reply src/test/resources/application-test.yml]: If we have a dedicated configuration file for a test (that's not a profile) I'd expect:

1. To only contain what is required
2. To not be named like it's a standard file that would be picked up if a `test` profile is active.

**Comment by snicoll** [comment]
Thanks for the PR. In the meantime, we've added a full integration tests suite that validates a lot more variables but also increases significantly the total build time. Given that 2.2 will improve this, the extra complexity in the build setup is not justified IMO.
